### PR TITLE
chore(crypto/bls12381): remove cosmos/crypto wrapper for BLS curve

### PR DESCRIPTION
**Pull request by melekes**
I don't think it adds much value; hence, we can use the [underlying library](https://github.com/supranational/blst/blob/master/bindings/go) directly. This PR is the preparation for the BLS signature aggregation.

* remove `cosmos/crypto` wrapper for BLS curve

---

#### PR checklist

- [x] Tests written/updated
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [ ] ~~Updated relevant documentation (`docs/` or `spec/`) and code comments~~
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by cason** [comment]
This failure on `TestMaxCommitBytes` is not a random stuff, right? Probably the size of the sigs on the block header have changed.

**Comment by cason** [comment]
Isn't this breaking right now? We are changing the size of the block headers, aren't we?

**Comment by melekes** [comment]
> Isn't this breaking right now? We are changing the size of the block headers, aren't we?

I will break this PR into 2.

File: crypto/bls12381/const.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ const (
 	// PrivKeySize defines the length of the PrivKey byte array.
 	PrivKeySize = 32
 	// PubKeySize defines the length of the PubKey byte array.
-	PubKeySize = 48
```
Comment (melekes) [inline crypto/bls12381/const.go:7]: It was not 48!

File: crypto/bls12381/key_bls12381.go [hunk 2]
```diff
@@ -18,6 +20,21 @@ const (
...
 )
 
+var (
+	// ErrDeserialization is returned when deserialization fails.
+	ErrDeserialization = errors.New("bls12381: deserialization error")
+
+	dstMinSig = []byte("BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_")
+)
+
+// For minimal-pubkey-size operations.
```
Comment (melekes) [inline crypto/bls12381/key_bls12381.go:30]: We need to change this to minimal-signature-size operations because we care about signature size more!

**Comment by cason** [comment]
Can you change the title to "remove cosmos/crypto wrapper for BLS curve"?

File: crypto/bls12381/key_bls12381.go (hunk 2 above)
Comment (cason) [reply crypto/bls12381/key_bls12381.go:29]: Can you open an issue for that?

**Review by cason — APPROVED** [review]
Legit.

File: crypto/bls12381/key_bls12381.go (hunk 2 above)
Comment (melekes) [reply crypto/bls12381/key_bls12381.go:29]: https://github.com/cometbft/cometbft/issues/3624

**Pull request merged**